clang/tools/driver/cc1depscand_main.cpp

```diff
@@ -669,7 +670,7 @@ int cc1depscand_main(ArrayRef<const char *> Argv, const char *Argv0,
         // FIXME: we re-compute the cache key here and try to access the action
         // cache and see if it should be a cache hit. Is there a better way to
         // get this stats in the daemon?
-        auto &CAS = FS->getCAS();
+        auto &CAS = ProxyFS->getCAS();
```

Comment:
What is this for? There's only one CAS.

Reply:
For consistency, so the original FS is not referred after the Proxy is created in the thread.

Reply:
I don't think adding `ProxyFS` in this function is worth it just for that.

Note that there is already a proxy for the thread created by the DependencyScanningWorker in its constructor. It'd be better to use that if you want to make things consistent.